java/integTest/src/test/java/compatibility/jedis/UnifiedJedisTest.java

```diff
@@ -0,0 +1,271 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package compatibility.jedis;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.*;
+import redis.clients.jedis.UnifiedJedis;
+
+/**
+ * UnifiedJedis compatibility test that validates GLIDE UnifiedJedis functionality.
+ *
+ * <p>This test ensures that the GLIDE compatibility layer provides the expected UnifiedJedis API
+ * and behavior for core Redis operations.
+ */
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class UnifiedJedisTest {
+
+    private static final String TEST_KEY_PREFIX = "unified_jedis_test:";
+
+    // Server configuration - dynamically resolved from CI environment
+    private static String redisHost;
+    private static int redisPort;
+
+    // GLIDE UnifiedJedis compatibility layer instance
+    private UnifiedJedis unifiedJedis;
+
+    @BeforeAll
+    static void setupClass() {
+        resolveServerAddress();
+    }
+
+    @BeforeEach
+    void setup() {
+        // Create GLIDE UnifiedJedis compatibility layer instance
+        try {
+            unifiedJedis = new UnifiedJedis(redisHost, redisPort);
+            assertNotNull(unifiedJedis, "GLIDE UnifiedJedis instance should be created successfully");
+        } catch (Exception e) {
+            fail("Failed to create GLIDE UnifiedJedis instance: " + e.getMessage());
```

Comment:
We should not use this pattern. Let the error propagate so that you can report stack traces properly.